Windows VM with Azure secure baseline

mgreenegit opened the pull request:
# PR Checklist

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

mgreenegit commented on quickstarts/microsoft.compute/vm-windows-baseline/main.bicep:
```diff
@@ -0,0 +1,224 @@
+@description('Enable accelerated networking.')
+param enableAcceleratedNetworking bool = true
+
+resource networkSecurityGroup 'Microsoft.Network/networkSecurityGroups@2021-05-01' = {
```
added rules @bmoore-msft

mgreenegit replied:
@bmoore-msft any task accomplished through means other than random human clicking ;-)

bmoore-msft replied:
Thinking from a QuickStart perspective... I spin up the VM, then what?  It doesn't appear to be running a workload so the only thing I would do next is to login to the VM and see what this hardening is all about?  And if 3389 isn't open how would I see what this thing is supposed to do for me?

mgreenegit replied:
We should dissuade people from that behavior. It is not OK to log in to servers just to see what's-what. There are great tools for that, from Azure Policy to run command.

It would be interesting to add port 22 for SSH (yes, on Windows) and add a configuration that enables PowerShell transcription. That is one way to have an interactive, rich, diagnostic experience without compromising security and reliability. That could detract from the intent of this quickstart, however.

bmoore-msft replied:
Either sounds fine - as a customer... what should I do after deploying this?  Can we add something to the readme along the lines of "After deploying this template you can..."?

mgreenegit replied:
Great idea. Pushed a new block of text.

mgreenegit commented:
#sign-off

mgreenegit commented on quickstarts/microsoft.compute/vm-windows-baseline/main.bicep:
```diff
@@ -1,5 +1,5 @@
 @description('The Azure region where resources in the template should be deployed.')
-param location string = resourceGroup().location
+param location string = 'eastus'
```
actually fixed it this time

mgreenegit commented on quickstarts/microsoft.compute/vm-windows-baseline/README.md:
```diff
@@ -0,0 +1,44 @@
+![Bicep Version](https://azurequickstartsservice.blob.core.windows.net/badges/quickstarts/microsoft.compute/vm-windows-baseline/BicepVersion.svg)
+
+[![Deploy To Azure](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/deploytoazure.svg?sanitize=true)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2Fquickstarts%2Fmicrosoft.compute%2Fvm-windows-baseline%2Fazuredeploy.json)
+[![Deploy To Azure US Gov](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/deploytoazuregov.svg?sanitize=true)](https://portal.azure.us/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2Fquickstarts%2Fmicrosoft.compute%2Fvm-windows-baseline%2Fazuredeploy.json)
```
this one should work in Gov

mgreenegit commented:
#sign-off @bmoore-msft

mgreenegit commented on quickstarts/microsoft.compute/vm-windows-baseline/azuredeploy.parameters.json:
```diff
@@ -0,0 +1,18 @@
+    "location": {
+      "value": "eastus"
+    },
+    "vmName": {
```
I didn't understand at first. I removed the default value. That way the person doing the deployment will always need to pick a name, and the test service can evaluate random names.

mgreenegit commented:
@bmoore-msft in the latest commit, I removed the default value for the VM.

also added comment about RDP: https://github.com/Azure/azure-quickstart-templates/pull/11931#discussion_r748745869

mgreenegit commented:
👍
